src/Uuid/Uuid.php

```diff
@@ -214,6 +189,43 @@ final public static function for(
 		};
 	}
 
+	/**
+	 * Creates a UUID object from a UUID string
+	 * @since 6.0.0
+	 */
+	final public static function from(
+		string $uuid,
+		string|array|null $scheme = null,
+		Collection|null $context = null
+	): static|null {
+		if (Uuids::enabled() === false) {
+			return null;
+		}
+
+		if ($scheme !== null && static::is($uuid, $scheme) === false) {
+			return null;
+		}
+
+		if ($uri = Str::before($uuid, '://')) {
+			return match ($uri) {
+				'page'   => new PageUuid(uuid: $uuid, context: $context),
+				'file'   => new FileUuid(uuid: $uuid, context: $context),
+				'site'   => new SiteUuid(uuid: $uuid, context: $context),
+				'user'   => new UserUuid(uuid: $uuid, context: $context),
+				// TODO: activate for uuid-block-structure-support
+				// 'block'  => new BlockUuid(uuid: $seed, context: $context),
+				// 'struct' => new StructureUuid(uuid: $seed, context: $context),
+				default  => throw new InvalidArgumentException(
+					message: 'Invalid UUID URI "' . $uri . '" in ' . $uuid
```

Comment:
```suggestion
					message: 'Invalid UUID type "' . $uri . '" in ' . $uuid
```

Probably makes sense to rename the variable as well. Not sure why we used `$uri` here.